File: swift_reference/Views/CircleSundaySummaryView.swift
import SwiftUI

struct CircleSundaySummaryView: View {
    let circleId: String
    let circleName: String

    @State private var summary: SundaySummaryResponse?
    @State private var isLoading: Bool = true
    @State private var error: String?
    @State private var gratitudeWeekCount: Int = 0

    @Environment(\.dismiss) private var dismiss

    var body: some View {
        NavigationStack {
            Group {
                if isLoading {
                    ProgressView()
                        .frame(maxWidth: .infinity, maxHeight: .infinity)
                } else if let summary {
                    summaryContent(summary)
                } else if let error {
                    VStack(spacing: 12) {
                        Image(systemName: "exclamationmark.triangle")
                            .font(.title)
                            .foregroundStyle(.secondary)
                        Text(error)
                            .font(.subheadline)
                            .foregroundStyle(.secondary)
                        Button("Retry") {
                            Task { await loadSummary() }
                        }
                        .foregroundStyle(TributeColor.golden)
                    }
                    .frame(maxWidth: .infinity, maxHeight: .infinity)
                }
            }
            .background {
                ZStack {
                    TributeColor.charcoal.ignoresSafeArea()
                    TributeColor.warmGlow.ignoresSafeArea()
                }
            }
            .navigationTitle("Weekly Summary")
            .navigationBarTitleDisplayMode(.inline)
            .toolbarColorScheme(.dark, for: .navigationBar)
            .toolbar {
                ToolbarItem(placement: .cancellationAction) {
                    Button("Done") { dismiss() }
                        .foregroundStyle(.secondary)
                }
            }
            .task {
                await loadSummary()
                await loadGratitudeCount()
            }
        }
    }

    @ViewBuilder
    private func summaryContent(_ summary: SundaySummaryResponse) -> some View {
        ScrollView {
            VStack(spacing: 28) {
                VStack(spacing: 12) {
                    ZStack {
                        Circle()
                            .fill(
                                RadialGradient(
                                    colors: [
                                        TributeColor.golden.opacity(0.2),
                                        TributeColor.golden.opacity(0.05)
                                    ],
                                    center: .center,
                                    startRadius: 0,
                                    endRadius: 50
                                )
                            )
                            .frame(width: 80, height: 80)
                        Image(systemName: "sun.max.fill")
                            .font(.system(size: 32))
                            .foregroundStyle(TributeColor.golden)
                    }

                    Text(circleName)
                        .font(.system(.title3, design: .serif, weight: .bold))
                        .foregroundStyle(TributeColor.warmWhite)

                    Text("This week's faithfulness")
                        .font(.system(.subheadline, design: .serif))
                        .foregroundStyle(TributeColor.softGold.opacity(0.7))
                }
                .padding(.top, 8)

                HStack(spacing: 0) {
                    statCard(
                        value: "\(summary.activeMembers)",
                        label: "Active",
                        sublabel: "of \(summary.totalMembers)",
                        color: TributeColor.sage
                    )
                    statCard(
                        value: "\(Int(summary.averageScore * 100))%",
                        label: "Avg Score",
                        sublabel: "this week",
                        color: TributeColor.golden
                    )
                    statCard(
                        value: "\(summary.totalMembers)",
                        label: "Members",
                        sublabel: "total",
                        color: TributeColor.softGold
                    )
                }
                .padding(.horizontal, 20)

                if summary.averageScore > 0 {
                    faithfulnessBar(score: summary.averageScore)
                        .padding(.horizontal, 20)
                }

                if !summary.topStreaks.isEmpty {
                    topStreaksSection(summary.topStreaks)
                        .padding(.horizontal, 20)
                }

                if gratitudeWeekCount > 0 {
                    gratitudeCountCard
                        .padding(.horizontal, 20)
                }

                VStack(spacing: 6) {
                    Text("\u{201C}Therefore encourage one another and build one another up, just as you are doing.\u{201D}")
                        .font(.system(.subheadline, design: .serif))
                        .italic()
                        .foregroundStyle(TributeColor.softGold.opacity(0.6))
                        .multilineTextAlignment(.center)
                    Text("1 Thessalonians 5:11")
                        .font(.caption)
                        .foregroundStyle(TributeColor.golden.opacity(0.5))
                }
                .padding(.horizontal, 32)
                .padding(.top, 8)
            }
            .padding(.bottom, 40)
        }
    }

    private func statCard(value: String, label: String, sublabel: String, color: Color) -> some View {
        VStack(spacing: 6) {
            Text(value)
                .font(.system(size: 28, weight: .bold, design: .rounded))
                .foregroundStyle(color)

            Text(label)
                .font(.caption.weight(.semibold))
                .foregroundStyle(TributeColor.warmWhite)

            Text(sublabel)
                .font(.caption2)
                .foregroundStyle(.secondary)
        }
        .frame(maxWidth: .infinity)
        .padding(.vertical, 16)
        .background(TributeColor.cardBackground)
        .clipShape(.rect(cornerRadius: 12))
        .overlay(
            RoundedRectangle(cornerRadius: 12)
                .strokeBorder(TributeColor.cardBorder, lineWidth: 0.5)
        )
        .padding(.horizontal, 4)
    }

    private func faithfulnessBar(score: Double) -> some View {
        VStack(alignment: .leading, spacing: 10) {
            Label("Circle Faithfulness", systemImage: "chart.bar.fill")
                .font(.caption.weight(.semibold))
                .foregroundStyle(TributeColor.golden)

            GeometryReader { geo in
                ZStack(alignment: .leading) {
                    RoundedRectangle(cornerRadius: 6)
                        .fill(TributeColor.cardBackground)
                        .frame(height: 12)

                    RoundedRectangle(cornerRadius: 6)
                        .fill(
                            LinearGradient(
                                colors: [TributeColor.golden.opacity(0.8), TributeColor.golden],
                                startPoint: .leading,
                                endPoint: .trailing
                            )
                        )
                        .frame(width: geo.size.width * min(score, 1.0), height: 12)
                }
            }
            .frame(height: 12)

            HStack {
                Text(scoreMessage(score))
                    .font(.system(.caption, design: .serif))
                    .foregroundStyle(.secondary)
                Spacer()
            }
        }
        .padding(16)
        .background(TributeColor.golden.opacity(0.04))
        .clipShape(.rect(cornerRadius: 14))
        .overlay(
            RoundedRectangle(cornerRadius: 14)
                .strokeBorder(TributeColor.golden.opacity(0.1), lineWidth: 0.5)
        )
    }

    private func topStreaksSection(_ streaks: [TopStreak]) -> some View {
        VStack(alignment: .leading, spacing: 12) {
            Label("Top Streaks", systemImage: "flame.fill")
                .font(.caption.weight(.semibold))
                .foregroundStyle(TributeColor.warmCoral)

            ForEach(Array(streaks.prefix(5).enumerated()), id: \.element.id) { index, streak in
                HStack(spacing: 12) {
                    Text("\(index + 1)")
                        .font(.caption2.weight(.bold))
                        .foregroundStyle(index == 0 ? TributeColor.golden : .secondary)
                        .frame(width: 20)

                    ZStack {
                        Circle()
                            .fill(index == 0 ? TributeColor.golden.opacity(0.12) : TributeColor.sage.opacity(0.1))
                            .frame(width: 28, height: 28)
                        Image(systemName: "person.fill")
                            .font(.system(size: 11))
                            .foregroundStyle(index == 0 ? TributeColor.golden : TributeColor.sage)
                    }

                    Text("Member")
                        .font(.subheadline)
                        .foregroundStyle(.primary)

                    Spacer()

                    HStack(spacing: 4) {
                        Image(systemName: "flame.fill")
                            .font(.caption2)
                            .foregroundStyle(TributeColor.warmCoral)
                        Text("\(streak.streak) days")
                            .font(.caption.weight(.medium))
                            .foregroundStyle(TributeColor.warmCoral)
                    }
                }
            }
        }
        .padding(16)
        .background(TributeColor.warmCoral.opacity(0.04))
        .clipShape(.rect(cornerRadius: 14))
        .overlay(
            RoundedRectangle(cornerRadius: 14)
                .strokeBorder(TributeColor.warmCoral.opacity(0.1), lineWidth: 0.5)
        )
    }

    private func scoreMessage(_ score: Double) -> String {
        if score >= 0.9 { return "Outstanding! Your circle walked in near-perfect faithfulness this week." }
        if score >= 0.7 { return "Strong week! Your circle showed up with consistency and dedication." }
        if score >= 0.5 { return "Good effort! More than half the circle stayed faithful this week." }
        if score >= 0.3 { return "A start! Every small step counts. Encourage each other." }
        return "A quiet week. Rally together — you're stronger in community."
    }

    private var gratitudeCountCard: some View {
        HStack(spacing: 12) {
            ZStack {
                Circle()
                    .fill(TributeColor.golden.opacity(0.12))
                    .frame(width: 36, height: 36)
                Image(systemName: "heart.text.clipboard")
                    .font(.system(size: 14))
                    .foregroundStyle(TributeColor.golden)
            }

            VStack(alignment: .leading, spacing: 2) {
                Text("Shared Gratitudes")
                    .font(.caption.weight(.semibold))
                    .foregroundStyle(TributeColor.warmWhite)
                Text("\(gratitudeWeekCount) gratitude\(gratitudeWeekCount == 1 ? "" : "s") shared this week")
                    .font(.caption)
                    .foregroundStyle(.secondary)
            }

            Spacer()

            Text("\(gratitudeWeekCount)")
                .font(.system(size: 22, weight: .bold, design: .rounded))
                .foregroundStyle(TributeColor.golden)
        }
        .padding(16)
        .background(TributeColor.golden.opacity(0.04))
        .clipShape(.rect(cornerRadius: 14))
        .overlay(
            RoundedRectangle(cornerRadius: 14)
                .strokeBorder(TributeColor.golden.opacity(0.1), lineWidth: 0.5)
        )
    }

    private func loadSummary() async {
        isLoading = true
        error = nil
        do {
            summary = try await APIService.shared.getSundaySummary(circleId: circleId)
        } catch {
            self.error = error.localizedDescription
        }
        isLoading = false
    }

    private func loadGratitudeCount() async {
        do {
            let response = try await APIService.shared.getGratitudeWeekCount(circleId: circleId)
            gratitudeWeekCount = response.weekCount
        } catch {}
    }
}

File: lib/views/circles/circle_sunday_summary_view.dart
import 'package:flutter/material.dart';
import '../../services/api_service.dart';
import '../../theme/app_theme.dart';

class CircleSundaySummaryView extends StatefulWidget {
  final String circleId;
  final String circleName;

  const CircleSundaySummaryView({super.key, required this.circleId, required this.circleName});

  @override
  State<CircleSundaySummaryView> createState() => _CircleSundaySummaryViewState();
}

class _CircleSundaySummaryViewState extends State<CircleSundaySummaryView> {
  SundaySummaryResponse? _summary;
  bool _isLoading = true;
  String? _error;
  int _gratitudeWeekCount = 0;

  @override
  void initState() {
    super.initState();
    _loadSummary();
    _loadGratitudeCount();
  }

  Future<void> _loadSummary() async {
    setState(() { _isLoading = true; _error = null; });
    try {
      final summary = await APIService.shared.getSundaySummary(widget.circleId);
      if (mounted) setState(() { _summary = summary; _isLoading = false; });
    } catch (e) {
      if (mounted) setState(() { _error = e.toString(); _isLoading = false; });
    }
  }

  Future<void> _loadGratitudeCount() async {
    try {
      final r = await APIService.shared.getGratitudeWeekCount(widget.circleId);
      if (mounted) setState(() => _gratitudeWeekCount = r.weekCount);
    } catch (_) {}
  }

  String _scoreMessage(double score) {
    if (score >= 0.9) return 'Outstanding! Your circle walked in near-perfect faithfulness this week.';
    if (score >= 0.7) return 'Strong week! Your circle showed up with consistency and dedication.';
    if (score >= 0.5) return 'Good effort! More than half the circle stayed faithful this week.';
    if (score >= 0.3) return 'A start! Every small step counts. Encourage each other.';
    return 'A quiet week. Rally together — you\'re stronger in community.';
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: TributeColor.charcoal,
      appBar: AppBar(
        backgroundColor: TributeColor.charcoal,
        title: const Text('Weekly Summary',
            style: TextStyle(color: TributeColor.warmWhite, fontSize: 17, fontWeight: FontWeight.w600)),
        actions: [
          TextButton(
            onPressed: () => Navigator.pop(context),
            child: Text('Done', style: TextStyle(color: Colors.white.withValues(alpha: 0.5))),
          ),
        ],
        automaticallyImplyLeading: false,
      ),
      body: SafeArea(child: _buildBody()),
    );
  }

  Widget _buildBody() {
    if (_isLoading) {
      return const Center(child: CircularProgressIndicator(color: TributeColor.golden));
    }
    if (_summary == null) {
      return Center(
        child: Column(mainAxisAlignment: MainAxisAlignment.center, children: [
          Icon(Icons.warning_amber_rounded, size: 32, color: Colors.white.withValues(alpha: 0.3)),
          const SizedBox(height: 12),
          Text(_error ?? 'Failed to load',
              style: TextStyle(color: Colors.white.withValues(alpha: 0.5))),
          const SizedBox(height: 16),
          TextButton(
            onPressed: _loadSummary,
            child: const Text('Retry', style: TextStyle(color: TributeColor.golden)),
          ),
        ]),
      );
    }
    return _summaryContent(_summary!);
  }

  Widget _summaryContent(SundaySummaryResponse s) {
    return SingleChildScrollView(
      padding: const EdgeInsets.fromLTRB(20, 8, 20, 40),
      child: Column(children: [
        const SizedBox(height: 8),
        Container(
          width: 80, height: 80,
          decoration: BoxDecoration(
            shape: BoxShape.circle,
            gradient: RadialGradient(colors: [
              TributeColor.golden.withValues(alpha: 0.2),
              TributeColor.golden.withValues(alpha: 0.05),
            ]),
          ),
          child: const Icon(Icons.wb_sunny_rounded, size: 32, color: TributeColor.golden),
        ),
        const SizedBox(height: 12),
        Text(widget.circleName,
            style: const TextStyle(fontSize: 20, fontWeight: FontWeight.w700, color: TributeColor.warmWhite)),
        const SizedBox(height: 4),
        Text("This week's faithfulness",
            style: TextStyle(fontSize: 14, color: TributeColor.softGold.withValues(alpha: 0.7))),
        const SizedBox(height: 24),
        Row(children: [
          Expanded(child: _statCard('${s.activeMembers}', 'Active', 'of ${s.totalMembers}', TributeColor.sage)),
          const SizedBox(width: 8),
          Expanded(child: _statCard('${(s.averageScore * 100).toInt()}%', 'Avg Score', 'this week', TributeColor.golden)),
          const SizedBox(width: 8),
          Expanded(child: _statCard('${s.totalMembers}', 'Members', 'total', TributeColor.softGold)),
        ]),
        if (s.averageScore > 0) ...[
          const SizedBox(height: 20),
          _faithfulnessBar(s.averageScore),
        ],
        if (s.topStreaks.isNotEmpty) ...[
          const SizedBox(height: 20),
          _topStreaksSection(s.topStreaks),
        ],
        if (_gratitudeWeekCount > 0) ...[
          const SizedBox(height: 20),
          _gratitudeCountCard(),
        ],
        const SizedBox(height: 28),
        Text(
          '\u201CTherefore encourage one another and build one another up, just as you are doing.\u201D',
          textAlign: TextAlign.center,
          style: TextStyle(fontSize: 14, fontStyle: FontStyle.italic,
              color: TributeColor.softGold.withValues(alpha: 0.6), height: 1.6),
        ),
        const SizedBox(height: 6),
        Text('1 Thessalonians 5:11',
            style: TextStyle(fontSize: 11, color: TributeColor.golden.withValues(alpha: 0.5))),
      ]),
    );
  }

  Widget _statCard(String value, String label, String sublabel, Color color) {
    return Container(
      padding: const EdgeInsets.symmetric(vertical: 16),
      decoration: BoxDecoration(
        color: TributeColor.cardBackground,
        borderRadius: BorderRadius.circular(12),
        border: Border.all(color: TributeColor.cardBorder, width: 0.5),
      ),
      child: Column(children: [
        Text(value, style: TextStyle(fontSize: 24, fontWeight: FontWeight.w700, color: color)),
        const SizedBox(height: 4),
        Text(label, style: const TextStyle(fontSize: 12, fontWeight: FontWeight.w600, color: TributeColor.warmWhite)),
        Text(sublabel, style: TextStyle(fontSize: 10, color: Colors.white.withValues(alpha: 0.4))),
      ]),
    );
  }

  Widget _faithfulnessBar(double score) {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: TributeColor.golden.withValues(alpha: 0.04),
        borderRadius: BorderRadius.circular(14),
        border: Border.all(color: TributeColor.golden.withValues(alpha: 0.1), width: 0.5),
      ),
      child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
        Row(children: [
          const Icon(Icons.bar_chart_rounded, size: 14, color: TributeColor.golden),
          const SizedBox(width: 6),
          const Text('Circle Faithfulness',
              style: TextStyle(fontSize: 12, fontWeight: FontWeight.w600, color: TributeColor.golden)),
        ]),
        const SizedBox(height: 10),
        LayoutBuilder(builder: (_, constraints) {
          return Stack(children: [
            Container(
              height: 12,
              decoration: BoxDecoration(
                  color: TributeColor.cardBackground, borderRadius: BorderRadius.circular(6)),
            ),
            Container(
              height: 12,
              width: constraints.maxWidth * score.clamp(0.0, 1.0),
              decoration: BoxDecoration(
                gradient: LinearGradient(
                  colors: [TributeColor.golden.withValues(alpha: 0.8), TributeColor.golden],
                ),
                borderRadius: BorderRadius.circular(6),
              ),
            ),
          ]);
        }),
        const SizedBox(height: 8),
        Text(_scoreMessage(score),
            style: TextStyle(fontSize: 12, color: Colors.white.withValues(alpha: 0.5))),
      ]),
    );
  }

  Widget _topStreaksSection(List<TopStreak> streaks) {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: TributeColor.warmCoral.withValues(alpha: 0.04),
        borderRadius: BorderRadius.circular(14),
        border: Border.all(color: TributeColor.warmCoral.withValues(alpha: 0.1), width: 0.5),
      ),
      child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
        const Row(children: [
          Icon(Icons.local_fire_department, size: 14, color: TributeColor.warmCoral),
          SizedBox(width: 6),
          Text('Top Streaks', style: TextStyle(fontSize: 12, fontWeight: FontWeight.w600, color: TributeColor.warmCoral)),
        ]),
        const SizedBox(height: 12),
        ...streaks.take(5).toList().asMap().entries.map((e) {
          final i = e.key;
          final streak = e.value;
          final isFirst = i == 0;
          return Padding(
            padding: const EdgeInsets.only(bottom: 10),
            child: Row(children: [
              SizedBox(
                width: 20,
                child: Text('${i + 1}',
                    style: TextStyle(fontSize: 11, fontWeight: FontWeight.w700,
                        color: isFirst ? TributeColor.golden : Colors.white.withValues(alpha: 0.4))),
              ),
              Container(
                width: 28, height: 28,
                decoration: BoxDecoration(
                    shape: BoxShape.circle,
                    color: isFirst
                        ? TributeColor.golden.withValues(alpha: 0.12)
                        : TributeColor.sage.withValues(alpha: 0.1)),
                child: Icon(Icons.person_rounded, size: 12,
                    color: isFirst ? TributeColor.golden : TributeColor.sage),
              ),
              const SizedBox(width: 10),
              const Expanded(
                child: Text('Member', style: TextStyle(fontSize: 14, color: TributeColor.warmWhite)),
              ),
              Row(children: [
                const Icon(Icons.local_fire_department, size: 11, color: TributeColor.warmCoral),
                const SizedBox(width: 4),
                Text('${streak.streak} days',
                    style: const TextStyle(fontSize: 12, fontWeight: FontWeight.w500, color: TributeColor.warmCoral)),
              ]),
            ]),
          );
        }),
      ]),
    );
  }

  Widget _gratitudeCountCard() {
    return Container(
      padding: const EdgeInsets.all(16),
      decoration: BoxDecoration(
        color: TributeColor.golden.withValues(alpha: 0.04),
        borderRadius: BorderRadius.circular(14),
        border: Border.all(color: TributeColor.golden.withValues(alpha: 0.1), width: 0.5),
      ),
      child: Row(children: [
        Container(
          width: 36, height: 36,
          decoration: BoxDecoration(
              shape: BoxShape.circle, color: TributeColor.golden.withValues(alpha: 0.12)),
          child: const Icon(Icons.favorite_rounded, size: 16, color: TributeColor.golden),
        ),
        const SizedBox(width: 12),
        Expanded(child: Column(crossAxisAlignment: CrossAxisAlignment.start, children: [
          const Text('Shared Gratitudes',
              style: TextStyle(fontSize: 12, fontWeight: FontWeight.w600, color: TributeColor.warmWhite)),
          Text(
            '$_gratitudeWeekCount gratitude${_gratitudeWeekCount == 1 ? '' : 's'} shared this week',
            style: TextStyle(fontSize: 12, color: Colors.white.withValues(alpha: 0.4)),
          ),
        ])),
        Text('$_gratitudeWeekCount',
            style: const TextStyle(fontSize: 22, fontWeight: FontWeight.w700, color: TributeColor.golden)),
      ]),
    );
  }
}
